### PR TITLE
[JENKINS-32301] Support more than one Reply-To Address

### PR DESCRIPTION
**Pull request by andresrc**
See [JENKINS-32301](https://issues.jenkins-ci.org/browse/JENKINS-32301)

Some additional refactoring of `MimeMessageBuilderTest` included.

@reviewbybees

### CODE REVIEW

File: src/main/java/jenkins/plugins/mailer/tasks/MimeMessageBuilder.java [hunk 1]
```diff
@@ -110,10 +118,27 @@ public MimeMessageBuilder setFrom(@Nonnull String from) {
...
+            // Done after to leave the current value untouched if there is a parsing error.
+            this.replyTo.clear();
+            this.replyTo.addAll(addresses);
+        } catch(UnsupportedEncodingException e) {
+            logError("Unable to parse Reply-To Addresses " + replyTo, e);
+        }
+        return this;
+    }
+
+    public MimeMessageBuilder addReplyTo(@Nonnull String replyTo) {
```
Comment (armfergom) [inline src/main/java/jenkins/plugins/mailer/tasks/MimeMessageBuilder.java]: Used only for testing, right?

File: src/main/java/jenkins/plugins/mailer/tasks/MimeMessageBuilder.java [hunk 2]
```diff
@@ -110,10 +118,27 @@ public MimeMessageBuilder setFrom(@Nonnull String from) {
     }
 
     public MimeMessageBuilder setReplyTo(@Nonnull String replyTo) {
-        this.replyTo = replyTo;
+        try {
```
Comment (oleg-nenashev) [inline src/main/java/jenkins/plugins/mailer/tasks/MimeMessageBuilder.java]: MimeMessageBuilder is single-thread, right? If no, the new code introduces concurrency issues

File: src/test/java/jenkins/plugins/mailer/tasks/MimeMessageBuilderTest.java [hunk 3]
```diff
@@ -164,4 +172,28 @@ public void test_addRecipients_tokenizer() throws Exception {
         Assert.assertEquals("tom.yyyy@gmail.com", recipients[1].toString());
         Assert.assertEquals("tom.zzzz@gmail.com", recipients[2].toString());
```
Comment (armfergom) [inline src/test/java/jenkins/plugins/mailer/tasks/MimeMessageBuilderTest.java]: You could use constants everywhere since you are doing it in other places, just to be consistent.

**Comment by armfergom** [comment]
LGTM :bee: This is more a new feature rather than a bug fix unless the documentation says it should be working.

File: src/main/java/jenkins/plugins/mailer/tasks/MimeMessageBuilder.java [hunk 4]
```diff
@@ -231,17 +265,28 @@ private void addRecipients(MimeMessage msg) throws UnsupportedEncodingException,
         addRecipients(msg, cc, Message.RecipientType.CC);
         addRecipients(msg, bcc, Message.RecipientType.BCC);
     }
+
     private void addRecipients(MimeMessage msg, Set<InternetAddress> recipientList, Message.RecipientType recipientType) throws UnsupportedEncodingException, MessagingException {
         if (recipientList.isEmpty()) {
             return;
         }
+        final Collection<InternetAddress> recipients = recipientFilter != null ? recipientFilter.apply(recipientList) : recipientList;
```
Comment (oleg-nenashev) [inline src/main/java/jenkins/plugins/mailer/tasks/MimeMessageBuilder.java]: Is this code related to the PR title? I doubt. Maybe a standalone bugfix

Comment (andresrc) [inline src/main/java/jenkins/plugins/mailer/tasks/MimeMessageBuilder.java]: It is a small refactor to share the code needed for the PR with the existing one.

File: src/test/java/jenkins/plugins/mailer/tasks/MimeMessageBuilderTest.java (hunk 3 above)
Comment (andresrc) [inline src/test/java/jenkins/plugins/mailer/tasks/MimeMessageBuilderTest.java]: I wanted to avoid "unrelated changes", will do a further clean up in another PR.

**Comment by andresrc** [comment]
Rebased and documented `MimeMessageBuilder` non-thread-safety.

**Comment by abayer** [comment]
:bee:

**Comment by andresrc** [comment]
@reviewbybees done
